android/source/VideoLocker/src/org/edx/mobile/view/NavigationFragment.java

```diff
@@ -60,38 +61,62 @@ public void onClick(View v) {
                 if(act instanceof MyCoursesListActivity){
                     //if MyCourses pressed when on MyCourse screen, close drawer
                     ((MyCoursesListActivity) act).closeDrawer();
-                }else if(act instanceof MyVideosTabActivity){
-                    ((MyVideosTabActivity) act).closeDrawer();
-                    Intent myCoursesIntent = new Intent(getActivity(), MyCoursesListActivity.class);
+                }else {
+                    ((BaseFragmentActivity) act).closeDrawer();
+                    Intent myCoursesIntent = new Intent(act, MyCoursesListActivity.class);
                     myCoursesIntent.addFlags(Intent.FLAG_ACTIVITY_REORDER_TO_FRONT);
-                    getActivity().startActivity(myCoursesIntent);
-                    getActivity().finish();
+                    act.startActivity(myCoursesIntent);
+                    act.finish();
                 }
             }
         });
 
         TextView my_videos_tv = (TextView) layout.findViewById(R.id.my_assets);
         my_videos_tv.setOnClickListener(new OnClickListener() {
-
             @Override
             public void onClick(View v) {
                 Activity act = getActivity();
 
+                Intent findCoursesIntent = new Intent(getActivity(),
+                        MyVideosTabActivity.class);
+                findCoursesIntent.addFlags(Intent.FLAG_ACTIVITY_REORDER_TO_FRONT);
                 if(act instanceof MyCoursesListActivity){
                     ((MyCoursesListActivity) act).closeDrawer();
-                    Intent myVideosIntent = new Intent(getActivity(), 
-                            MyVideosTabActivity.class);
-                    myVideosIntent.addFlags(Intent.FLAG_ACTIVITY_REORDER_TO_FRONT);
-                    getActivity().startActivity(myVideosIntent);
+                    act.startActivity(findCoursesIntent);
                     //Finish is not called because the MyCourse activity need not be deleted
                 }else if(act instanceof MyVideosTabActivity){
                     ((MyVideosTabActivity) act).closeDrawer();
+                }else if(act instanceof FindCoursesActivity){
```

Comment:
This should probably just be an else to safely cover anything besides ``FindCoursesActivity`` that way it won't have to be touched when there's a new tab.